### PR TITLE
remove: clean gitignore content

### LINKED ISSUE #4013: dvc remove'd folder stays in gitignore
### Please provide information about your setup
The removed folder's name is still in `.gitignore`. Not sure if that's intentional or not
**Output of `dvc version`:**

```console
$ dvc version

DVC version: 0.93.0
Python version: 3.7.5
Platform: Darwin-19.5.0-x86_64-i386-64bit
Binary: True
Package: osxpkg
Supported remotes: azure, gdrive, gs, hdfs, http, https, s3, ssh, oss
Cache: reflink - supported, hardlink - supported, symlink - supported
Filesystem type (cache directory): ('apfs', '/dev/disk1s5')
Repo: dvc, git
Filesystem type (workspace): ('apfs', '/dev/disk1s5')
```

**Additional Information (if any):**

If applicable, please also provide a `--verbose` output of the command, eg: `dvc add --verbose`.

### PR DESCRIPTION
**Pull request by nik123**
Fixes #4013 

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

PR doesn't require documentation updates

### CODE REVIEW

File: dvc/output/base.py [hunk 1]
```diff
@@ -239,6 +239,12 @@ def ignore(self):
 
         self.repo.scm.ignore(self.fspath)
 
+    def ignore_remove(self):
```
Comment (efiop) [inline dvc/output/base.py:242]: Let's use this in `def remove()` down below too.

File: dvc/output/base.py [hunk 2]
```diff
@@ -337,7 +343,7 @@ def remove(self, ignore_remove=False):
             return
 
         if ignore_remove and self.use_scm_ignore:
```
Comment (efiop) [inline dvc/output/base.py:345]:
```suggestion
        if ignore_remove:
```

**Comment by jorgeorpinel** [comment]
Maybe this could be noted in https://dvc.org/doc/command-reference/remove instead of "removes all of its outputs (outs field),": "deletes all of its output files or directories (`outs` field), removes them from `.gitignore`, and optionally"...

**Comment by jorgeorpinel** [comment]
p.s. unrelated question on `remove`: If this is the only stage in dvc.yaml does dvc.yaml remain empty, or does it get removed? (along with dvc.lock)

**Comment by nik123** [comment]
> This could be noted in https://dvc.org/doc/command-reference/remove. Instead of "removes all of its outputs (outs field), and optionally...":
> 
> "deletes all of its output files or directories (outs field), removes them from .gitignore, and optionally..."

@jorgeorpinel should description for `dvc remove -h` be updated as well? If so, what should go first - updates in core repo or in documentation?

**Comment by efiop** [comment]
> p.s. unrelated question on remove @efiop: If this is the only stage in dvc.yaml does dvc.yaml remain empty, or does it get removed? (along with dvc.lock). Thanks

It is the easiest to just try running that and seeing :) But yes, both remain empty. Could clean that up a bit, for sure. Created https://github.com/iterative/dvc/issues/4090

**Comment by efiop** [comment]
> l should description for dvc remove -h be updated as well? If so, what should go first - updates in core repo or in documentation?

@nik123 Could start with core repo. Thanks! :pray:

**Comment by nik123** [comment]
@efiop , @jorgeorpinel : I checked cmd ref and ran into other problems which I think should be addressed before `.gitignore`. For details see #4094

**Comment by jorgeorpinel** [comment]
> should description for dvc remove -h be updated as well?
> checked cmd ref and ran into other problems

@nik123 thanks! Yes, if the core repo needs an update to the help output that would come first, and then update everything on the docs side 🙂

**Pull request merged**